FRW/Services/Manager/WalletManager.swift

```diff
@@ -796,6 +800,41 @@ extension WalletManager {
         }
     }
 
+    func fetchAccountInfo() async throws {
+        do {
+            let accountInfo = try await FlowNetwork.checkAccountInfo()
+            await MainActor.run {
+                self.accountInfo = accountInfo
+            }
+        } catch let error {
+            log.error("[EVM] fetch account info failed.\(error.localizedDescription)")
+            throw error
+        }
+    }
+    
+    var minimumStorageBalance: Decimal {
+        guard let accountInfo else { return Self.minDefaultBlance }
+        return accountInfo.storageFlow * Self.minDefaultBlance;
+    }
+    
+    var isStorageInsufficient: Bool {
+        return isStorageInsufficient(for: 0)
+    }
+
+    func isStorageInsufficient(for amount: Decimal) -> Bool {
+        guard let accountInfo else { return false }
+        return accountInfo.availableBalance - amount < Self.minDefaultBlance
```

Comment:
Storage check doesn't need amount, it should be `storageCapacity - storageUsed < 10000`

Reply:
Yeah I think I forgot to update it after we discussed it... btw I'm reimplementing this part using your pseudo-code